src/datadoc/frontend/components/resources_test_new_variables.py

```diff
@@ -2,120 +2,82 @@
+from datadoc.frontend.fields.display_new_variables import DisplayNewVariablesMetadata
 
-# Test data
-variables_test_names = [
-    "fnr",
-    "sivilstand",
-    "bostedskommune",
-    "inntekt",
-    "bankinnskudd",
-    "dato",
-]
+info_section = (
+    "Informasjon om hvordan jobbe i Datadoc, antall variabler i datasettet: osv.."
+)
 
-variables_test_names_dropdown = [
-    {"id": "1", "title": "fnr"},
-    {"id": "2", "title": "sivilstand"},
-    {"id": "3", "title": "bostedskommune"},
-    {"id": "4", "title": "inntekt"},
-    {"id": "5", "title": "bankinnskudd"},
-    {"id": "6", "title": "dato"},
-]
-# dropdown: datatype,variabelens, DPI,temporalitetstype,
-variables_details_test_obligatory = [
-    "Kortnavn",
-    "Navn",
-    "Datatype",
-    "Variabelens rolle",
-    "Definition URI",
-    "DPI",
-]
+NEW_DATASET_METADATA_INPUT = "new-dataset-metadata-input"
 
-data_type = [
-    {
-        "title": "Tekst",
-        "id": "text",
-    },
-    {
-        "title": "Heltall",
-        "id": "integer",
-    },
-    {
-        "title": "Desimaltall",
-        "id": "float",
-    },
-    {
-        "title": "Datotid",
-        "id": "datetime",
-    },
-    {
-        "title": "Boolsk",
-        "id": "bool",
-    },
-]
 
-variables_details_test_recommended = [
-    "Nøkkelord",
-    "Geografisk dekningsområde",
-    "Datakilde",
-    "Populasjonen",
-    "Kommentar",
-    "Temporalitetstype",
-    "Måleenhet",
-    "Format",
-    "Kodeverkets URI",
-    "Spesialverdienes URI",
-    "Ugyldige verdier",
-    "Inneholder data f.o.m.",
-    "Inneholder data t.o.m",
-]
-
-info_section = f"Informasjon om hvordan jobbe i Datadoc, antall variabler i datasettet: {len(variables_test_names)} osv.."
-
-
-# builders
-def build_input_dropdowns(name: str, values: list) -> ssb.Dropdown:
-    """Create dropdown."""
-    return ssb.Dropdown(
-        id=name,
-        header=name,
-        items=values,
-        className="variabels-dropdown",
-    )
-
-
-def build_input_field_section(variabel_list: list, id_type: str) -> html.Section:
+def build_input_field_section(
+    metadata_inputs: list[DisplayNewVariablesMetadata],
+) -> dbc.Form:
     """Create input fields."""
-    return html.Section(
+    return dbc.Form(
         [
-            html.Section(
-                [
-                    ssb.Input(
-                        label=value,
-                        id={"type": id_type, "id": str(index)},
-                        className="variabels-input",
-                    )
-                    for index, value in enumerate(variabel_list)
-                ],
-                className="variables-input-group",
-            ),
-            build_input_dropdowns("Datatype", data_type),
+            (
+                i.component(
+                    label=i.display_name,
+                    disabled=not i.editable,
+                    className="variabels-input",
+                    id={
+                        "type": NEW_DATASET_METADATA_INPUT,
```

Comment:
Should we not also use the name of the variable in the `id` dict? I think that's what has caused the problems with duplicate IDs.

Reply:
Here I already use name as part of the ID (i.identifier)